{"content": {"suites": {"0": {"status": {"total_pass": 1, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 2, "total_error": 0}, "tests": {"0": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"add-to-cart-sauce-labs-backpack\"]\"}\n     (Session info: chrome=134.0.6998.118); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF7181CFE65+26629]\n   \t(No symbol) [0x00007FF718136030]\n   \t(No symbol) [0x00007FF717FC931A]\n   \t(No symbol) [0x00007FF71801F8E7]\n   \t(No symbol) [0x00007FF71801FB1C]\n   \t(No symbol) [0x00007FF7180734A7]\n   \t(No symbol) [0x00007FF718047AEF]\n   \t(No symbol) [0x00007FF718070169]\n   \t(No symbol) [0x00007FF718047883]\n   \t(No symbol) [0x00007FF718010550]\n   \t(No symbol) [0x00007FF718011803]\n   \tGetHandleVerifier [0x00007FF7185272DD+3529853]\n   \tGetHandleVerifier [0x00007FF71853DA42+3621858]\n   \tGetHandleVerifier [0x00007FF7185324F3+3575443]\n   \tGetHandleVerifier [0x00007FF71829B79A+860474]\n   \t(No symbol) [0x00007FF7181408AF]\n   \t(No symbol) [0x00007FF71813CBE4]\n   \t(No symbol) [0x00007FF71813CD86]\n   \t(No symbol) [0x00007FF71812C2E9]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_run[chrome]", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test_run[firefox]", "rerun": "0"}, "2": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"add-to-cart-sauce-labs-backpack\"]\"}\n     (Session info: MicrosoftEdge=134.0.3124.72); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF75DD5A785+20677]\n   \tsimdutf::get_active_implementation [0x00007FF75DCB6F30+492592]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DFADC3A+1807418]\n   \t(No symbol) [0x00007FF75DAF72AC]\n   \t(No symbol) [0x00007FF75DAF758B]\n   \t(No symbol) [0x00007FF75DB385A7]\n   \t(No symbol) [0x00007FF75DB1849F]\n   \t(No symbol) [0x00007FF75DAED6ED]\n   \t(No symbol) [0x00007FF75DB35D61]\n   \t(No symbol) [0x00007FF75DB181C3]\n   \t(No symbol) [0x00007FF75DAECC36]\n   \t(No symbol) [0x00007FF75DAEC0C4]\n   \t(No symbol) [0x00007FF75DAECA63]\n   \t(No symbol) [0x00007FF75DBD58DD]\n   \t(No symbol) [0x00007FF75DBE1D42]\n   \t(No symbol) [0x00007FF75DBDA703]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DE3478A+262026]\n   \tsimdutf::get_active_implementation [0x00007FF75DCC4661+547681]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBD914+519700]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBDA59+520025]\n   \tsimdutf::get_active_implementation [0x00007FF75DCAF606+461574]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_run[edge]", "rerun": "0"}}, "suite_name": "e_commerce/test_saucedemo.py"}, "1": {"suite_name": "test_pages/test_changepassword.py", "tests": {"0": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"input-email\"]\"}\n     (Session info: chrome=134.0.6998.118); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF7181CFE65+26629]\n   \t(No symbol) [0x00007FF718136030]\n   \t(No symbol) [0x00007FF717FC931A]\n   \t(No symbol) [0x00007FF71801F8E7]\n   \t(No symbol) [0x00007FF71801FB1C]\n   \t(No symbol) [0x00007FF7180734A7]\n   \t(No symbol) [0x00007FF718047AEF]\n   \t(No symbol) [0x00007FF718070169]\n   \t(No symbol) [0x00007FF718047883]\n   \t(No symbol) [0x00007FF718010550]\n   \t(No symbol) [0x00007FF718011803]\n   \tGetHandleVerifier [0x00007FF7185272DD+3529853]\n   \tGetHandleVerifier [0x00007FF71853DA42+3621858]\n   \tGetHandleVerifier [0x00007FF7185324F3+3575443]\n   \tGetHandleVerifier [0x00007FF71829B79A+860474]\n   \t(No symbol) [0x00007FF7181408AF]\n   \t(No symbol) [0x00007FF71813CBE4]\n   \t(No symbol) [0x00007FF71813CD86]\n   \t(No symbol) [0x00007FF71812C2E9]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_chainging_password[chrome]", "rerun": "0"}, "1": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: [id=\"input-email\"]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\n   WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:197:5\n   NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:527:5\n   dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n", "test_name": "test_chainging_password[firefox]", "rerun": "0"}, "2": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"input-email\"]\"}\n     (Session info: MicrosoftEdge=134.0.3124.72); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF75DD5A785+20677]\n   \tsimdutf::get_active_implementation [0x00007FF75DCB6F30+492592]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DFADC3A+1807418]\n   \t(No symbol) [0x00007FF75DAF72AC]\n   \t(No symbol) [0x00007FF75DAF758B]\n   \t(No symbol) [0x00007FF75DB385A7]\n   \t(No symbol) [0x00007FF75DB1849F]\n   \t(No symbol) [0x00007FF75DAED6ED]\n   \t(No symbol) [0x00007FF75DB35D61]\n   \t(No symbol) [0x00007FF75DB181C3]\n   \t(No symbol) [0x00007FF75DAECC36]\n   \t(No symbol) [0x00007FF75DAEC0C4]\n   \t(No symbol) [0x00007FF75DAECA63]\n   \t(No symbol) [0x00007FF75DBD58DD]\n   \t(No symbol) [0x00007FF75DBE1D42]\n   \t(No symbol) [0x00007FF75DBDA703]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DE3478A+262026]\n   \tsimdutf::get_active_implementation [0x00007FF75DCC4661+547681]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBD914+519700]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBDA59+520025]\n   \tsimdutf::get_active_implementation [0x00007FF75DCAF606+461574]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_chainging_password[edge]", "rerun": "0"}}, "status": {"total_pass": 0, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 3, "total_error": 0}}, "2": {"suite_name": "test_pages/test_script.py", "tests": {"0": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"input-email\"]\"}\n     (Session info: chrome=134.0.6998.118); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF7181CFE65+26629]\n   \t(No symbol) [0x00007FF718136030]\n   \t(No symbol) [0x00007FF717FC931A]\n   \t(No symbol) [0x00007FF71801F8E7]\n   \t(No symbol) [0x00007FF71801FB1C]\n   \t(No symbol) [0x00007FF7180734A7]\n   \t(No symbol) [0x00007FF718047AEF]\n   \t(No symbol) [0x00007FF718070169]\n   \t(No symbol) [0x00007FF718047883]\n   \t(No symbol) [0x00007FF718010550]\n   \t(No symbol) [0x00007FF718011803]\n   \tGetHandleVerifier [0x00007FF7185272DD+3529853]\n   \tGetHandleVerifier [0x00007FF71853DA42+3621858]\n   \tGetHandleVerifier [0x00007FF7185324F3+3575443]\n   \tGetHandleVerifier [0x00007FF71829B79A+860474]\n   \t(No symbol) [0x00007FF7181408AF]\n   \t(No symbol) [0x00007FF71813CBE4]\n   \t(No symbol) [0x00007FF71813CD86]\n   \t(No symbol) [0x00007FF71812C2E9]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_validlogin_credentials[chrome]", "rerun": "0"}, "1": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: [id=\"input-email\"]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\n   WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:197:5\n   NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:527:5\n   dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n", "test_name": "test_validlogin_credentials[firefox]", "rerun": "0"}, "2": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"input-email\"]\"}\n     (Session info: MicrosoftEdge=134.0.3124.72); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF75DD5A785+20677]\n   \tsimdutf::get_active_implementation [0x00007FF75DCB6F30+492592]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DFADC3A+1807418]\n   \t(No symbol) [0x00007FF75DAF72AC]\n   \t(No symbol) [0x00007FF75DAF758B]\n   \t(No symbol) [0x00007FF75DB385A7]\n   \t(No symbol) [0x00007FF75DB1849F]\n   \t(No symbol) [0x00007FF75DAED6ED]\n   \t(No symbol) [0x00007FF75DB35D61]\n   \t(No symbol) [0x00007FF75DB181C3]\n   \t(No symbol) [0x00007FF75DAECC36]\n   \t(No symbol) [0x00007FF75DAEC0C4]\n   \t(No symbol) [0x00007FF75DAECA63]\n   \t(No symbol) [0x00007FF75DBD58DD]\n   \t(No symbol) [0x00007FF75DBE1D42]\n   \t(No symbol) [0x00007FF75DBDA703]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DE3478A+262026]\n   \tsimdutf::get_active_implementation [0x00007FF75DCC4661+547681]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBD914+519700]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBDA59+520025]\n   \tsimdutf::get_active_implementation [0x00007FF75DCAF606+461574]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_validlogin_credentials[edge]", "rerun": "0"}, "3": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"input-email\"]\"}\n     (Session info: chrome=134.0.6998.118); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF7181CFE65+26629]\n   \t(No symbol) [0x00007FF718136030]\n   \t(No symbol) [0x00007FF717FC931A]\n   \t(No symbol) [0x00007FF71801F8E7]\n   \t(No symbol) [0x00007FF71801FB1C]\n   \t(No symbol) [0x00007FF7180734A7]\n   \t(No symbol) [0x00007FF718047AEF]\n   \t(No symbol) [0x00007FF718070169]\n   \t(No symbol) [0x00007FF718047883]\n   \t(No symbol) [0x00007FF718010550]\n   \t(No symbol) [0x00007FF718011803]\n   \tGetHandleVerifier [0x00007FF7185272DD+3529853]\n   \tGetHandleVerifier [0x00007FF71853DA42+3621858]\n   \tGetHandleVerifier [0x00007FF7185324F3+3575443]\n   \tGetHandleVerifier [0x00007FF71829B79A+860474]\n   \t(No symbol) [0x00007FF7181408AF]\n   \t(No symbol) [0x00007FF71813CBE4]\n   \t(No symbol) [0x00007FF71813CD86]\n   \t(No symbol) [0x00007FF71812C2E9]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_invalidlogin_credentials[chrome]", "rerun": "0"}, "4": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: [id=\"input-email\"]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\n   WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:197:5\n   NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:527:5\n   dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n", "test_name": "test_invalidlogin_credentials[firefox]", "rerun": "0"}, "5": {"status": "FAIL", "message": "   selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"input-email\"]\"}\n     (Session info: MicrosoftEdge=134.0.3124.72); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF75DD5A785+20677]\n   \tsimdutf::get_active_implementation [0x00007FF75DCB6F30+492592]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DFADC3A+1807418]\n   \t(No symbol) [0x00007FF75DAF72AC]\n   \t(No symbol) [0x00007FF75DAF758B]\n   \t(No symbol) [0x00007FF75DB385A7]\n   \t(No symbol) [0x00007FF75DB1849F]\n   \t(No symbol) [0x00007FF75DAED6ED]\n   \t(No symbol) [0x00007FF75DB35D61]\n   \t(No symbol) [0x00007FF75DB181C3]\n   \t(No symbol) [0x00007FF75DAECC36]\n   \t(No symbol) [0x00007FF75DAEC0C4]\n   \t(No symbol) [0x00007FF75DAECA63]\n   \t(No symbol) [0x00007FF75DBD58DD]\n   \t(No symbol) [0x00007FF75DBE1D42]\n   \t(No symbol) [0x00007FF75DBDA703]\n   \tMicrosoft::Applications::Events::EventProperty::to_string [0x00007FF75DE3478A+262026]\n   \tsimdutf::get_active_implementation [0x00007FF75DCC4661+547681]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBD914+519700]\n   \tsimdutf::get_active_implementation [0x00007FF75DCBDA59+520025]\n   \tsimdutf::get_active_implementation [0x00007FF75DCAF606+461574]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_invalidlogin_credentials[edge]", "rerun": "0"}}, "status": {"total_pass": 0, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 6, "total_error": 0}}, "3": {"suite_name": "tutorial/test_assertion.py", "tests": {"0": {"status": "FAIL", "message": "E   AssertionError: ++++ soft assert failure details follow below ++++\nE   \nE   ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\nE   The following 2 failures were found in \"test_assertionsofttest\" (First_Automation_Tutorial.tutorial.test_assertion.Testassertion):\nE   +----------------------+----------------------+----------------------+\nFailure 1 (\"Verify gender, title and age_group\" method)\nE   +----------------------+----------------------+----------------------+\nE   Traceback (most recent call last):\n File \"C:\\Users\\AgnesYeboaaAnsah\\Documents\\Python_Selenium_Automation\\First_Automation_Tutorial\\tutorial\\test_assertion.py\", line 23, in test_assertionsofttest\n   self.soft_assert(self.assertIs, \"Female\", gender, \"gender is not female\")\n File \"C:\\Users\\AgnesYeboaaAnsah\\.venv\\Lib\\site-packages\\softest\\case.py\", line 62, in soft_assert\n   assert_method(*arguments, **keywords)\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 1166, in assertIs\n   self.fail(self._formatMessage(msg, standardMsg))\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 715, in fail\n   raise self.failureException(msg)\nE   AssertionError: 'Female' is not 'Female' : gender is not female\nE   \nE   -+ [1/2] +-\nE   \nE   +----------------------+----------------------+----------------------+\nFailure 2 (\"Verify gender, title and age_group\" method)\nE   +----------------------+----------------------+----------------------+\nE   Traceback (most recent call last):\n File \"C:\\Users\\AgnesYeboaaAnsah\\Documents\\Python_Selenium_Automation\\First_Automation_Tutorial\\tutorial\\test_assertion.py\", line 25, in test_assertionsofttest\n   self.soft_assert(self.assertIn, \"51\", age_group, \"51 is not in aged group\") # this message appears only when the condition fails\n File \"C:\\Users\\AgnesYeboaaAnsah\\.venv\\Lib\\site-packages\\softest\\case.py\", line 62, in soft_assert\n   assert_method(*arguments, **keywords)\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 1152, in assertIn\n   self.fail(self._formatMessage(msg, standardMsg))\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 715, in fail\n   raise self.failureException(msg)\nE   AssertionError: '51' not found in '5 - 15' : 51 is not in aged group\nE   \nE   -+ [2/2] +-\n", "test_name": "test_assertionsofttest", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test_firstassertion", "rerun": "0"}, "2": {"status": "FAIL", "message": "   AssertionError: gender is not female\n   assert 'Female' is 'Female'\n", "test_name": "test_assertionerror", "rerun": "0"}}, "status": {"total_pass": 1, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 2, "total_error": 0}}, "4": {"suite_name": "tutorial/test_assertionproject.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test_hardassert_project", "rerun": "0"}, "1": {"status": "FAIL", "message": "E   AssertionError: ++++ soft assert failure details follow below ++++\nE   \nE   ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\nE   The following 2 failures were found in \"test_softassert\" (First_Automation_Tutorial.tutorial.test_assertionproject.Testsoftassertproject):\nE   +----------------------+----------------------+----------------------+\nFailure 1 (\"Verify Warning message, title and Users\" method)\nE   +----------------------+----------------------+----------------------+\nE   Traceback (most recent call last):\n File \"C:\\Users\\AgnesYeboaaAnsah\\Documents\\Python_Selenium_Automation\\First_Automation_Tutorial\\tutorial\\test_assertionproject.py\", line 39, in test_softassert\n   self.soft_assert(self.assertIs, \"Epic sadface: Username and password do not match any user in this service\", warning_message, \"not the warning message\") # the identity is not the same that is why it give error.\n File \"C:\\Users\\AgnesYeboaaAnsah\\.venv\\Lib\\site-packages\\softest\\case.py\", line 62, in soft_assert\n   assert_method(*arguments, **keywords)\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 1166, in assertIs\n   self.fail(self._formatMessage(msg, standardMsg))\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 715, in fail\n   raise self.failureException(msg)\nE   AssertionError: 'Epic sadface: Username and password do not match any user in this service' is not 'Epic sadface: Username and password do not match any user in this service' : not the warning message\nE   \nE   -+ [1/2] +-\nE   \nE   +----------------------+----------------------+----------------------+\nFailure 2 (\"Verify Warning message, title and Users\" method)\nE   +----------------------+----------------------+----------------------+\nE   Traceback (most recent call last):\n File \"C:\\Users\\AgnesYeboaaAnsah\\Documents\\Python_Selenium_Automation\\First_Automation_Tutorial\\tutorial\\test_assertionproject.py\", line 41, in test_softassert\n   self.soft_assert(self.assertIn, \"Users\", warning_message, \"Users is not in the warning message\")\n File \"C:\\Users\\AgnesYeboaaAnsah\\.venv\\Lib\\site-packages\\softest\\case.py\", line 62, in soft_assert\n   assert_method(*arguments, **keywords)\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 1152, in assertIn\n   self.fail(self._formatMessage(msg, standardMsg))\n File \"C:\\Users\\AgnesYeboaaAnsah\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\unittest\\case.py\", line 715, in fail\n   raise self.failureException(msg)\nE   AssertionError: 'Users' not found in 'Epic sadface: Username and password do not match any user in this service' : Users is not in the warning message\nE   \nE   -+ [2/2] +-\n", "test_name": "test_softassert", "rerun": "0"}}, "status": {"total_pass": 1, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 1, "total_error": 0}}, "5": {"suite_name": "tutorial/test_failtest.py", "tests": {"0": {"status": "xFAIL", "message": "   assert 40 == (20 / 20)\n", "test_name": "test_divide", "rerun": "0"}, "1": {"status": "xFAIL", "message": "   assert 400 == (20 ** 3)\n", "test_name": "test_cube", "rerun": "0"}, "2": {"status": "xPASS", "message": "", "test_name": "test_square", "rerun": "0"}, "3": {"status": "xPASS", "message": "", "test_name": "test_sum", "rerun": "0"}, "4": {"status": "xFAIL", "message": "       _pytest.outcomes.XFailed: [NOTRUN]\n", "test_name": "test_sum2", "rerun": "0"}}, "status": {"total_pass": 0, "total_skip": 0, "total_xpass": 2, "total_xfail": 3, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "6": {"suite_name": "tutorial/test_firstproject.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test_first_automation", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test2_second_automation", "rerun": "0"}}, "status": {"total_pass": 2, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "7": {"suite_name": "tutorial/test_fixtures.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test1_order_history_title", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test2_change_password_title", "rerun": "0"}}, "status": {"total_pass": 2, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "8": {"suite_name": "tutorial/test_markers_run.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test_markersdemo", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test_endproccess", "rerun": "0"}, "2": {"status": "PASS", "message": "", "test_name": "test_login", "rerun": "0"}, "3": {"status": "PASS", "message": "", "test_name": "test_logout", "rerun": "0"}}, "status": {"total_pass": 4, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "9": {"suite_name": "tutorial/test_parametrization1.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test_param_decotator1[60-40-100]", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test_param_decotator1[30-20-50]", "rerun": "0"}, "2": {"status": "FAIL", "message": "   AssertionError: the sum total does not match the expected total\n   assert '60' == '50'\n     \n     - 50\n     + 60\n", "test_name": "test_param_decotator1[40-20-50]", "rerun": "0"}, "3": {"status": "PASS", "message": "", "test_name": "test_exponent[1-2]", "rerun": "0"}, "4": {"status": "PASS", "message": "", "test_name": "test_exponent[1-4]", "rerun": "0"}, "5": {"status": "PASS", "message": "", "test_name": "test_exponent[1-6]", "rerun": "0"}, "6": {"status": "PASS", "message": "", "test_name": "test_exponent[2-2]", "rerun": "0"}, "7": {"status": "PASS", "message": "", "test_name": "test_exponent[2-4]", "rerun": "0"}, "8": {"status": "PASS", "message": "", "test_name": "test_exponent[2-6]", "rerun": "0"}, "9": {"status": "PASS", "message": "", "test_name": "test_exponent[3-2]", "rerun": "0"}, "10": {"status": "PASS", "message": "", "test_name": "test_exponent[3-4]", "rerun": "0"}, "11": {"status": "PASS", "message": "", "test_name": "test_exponent[3-6]", "rerun": "0"}}, "status": {"total_pass": 11, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 1, "total_error": 0}}, "10": {"suite_name": "tutorial/test_parametrization2.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test_runtest[chrome]", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test_runtest[firefox]", "rerun": "0"}, "2": {"status": "PASS", "message": "", "test_name": "test_runtest[edge]", "rerun": "0"}}, "status": {"total_pass": 3, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "11": {"suite_name": "tutorial/test_practice.py", "tests": {"0": {"status": "FAIL", "message": "   selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\n     (Session info: chrome=134.0.6998.118); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\n   Stacktrace:\n   \tGetHandleVerifier [0x00007FF7181CFE65+26629]\n   \t(No symbol) [0x00007FF718136030]\n   \t(No symbol) [0x00007FF717FC931A]\n   \t(No symbol) [0x00007FF717FD0C7C]\n   \t(No symbol) [0x00007FF717FD3A3C]\n   \t(No symbol) [0x00007FF717FD3B0F]\n   \t(No symbol) [0x00007FF7180199B6]\n   \t(No symbol) [0x00007FF718047AAA]\n   \t(No symbol) [0x00007FF718012156]\n   \t(No symbol) [0x00007FF718047CC0]\n   \t(No symbol) [0x00007FF718070169]\n   \t(No symbol) [0x00007FF718047883]\n   \t(No symbol) [0x00007FF718010550]\n   \t(No symbol) [0x00007FF718011803]\n   \tGetHandleVerifier [0x00007FF7185272DD+3529853]\n   \tGetHandleVerifier [0x00007FF71853DA42+3621858]\n   \tGetHandleVerifier [0x00007FF7185324F3+3575443]\n   \tGetHandleVerifier [0x00007FF71829B79A+860474]\n   \t(No symbol) [0x00007FF7181408AF]\n   \t(No symbol) [0x00007FF71813CBE4]\n   \t(No symbol) [0x00007FF71813CD86]\n   \t(No symbol) [0x00007FF71812C2E9]\n   \tBaseThreadInitThunk [0x00007FFF14765550+16]\n   \tRtlUserThreadStart [0x00007FFF15D4485B+43]\n", "test_name": "test_me", "rerun": "0"}}, "status": {"total_pass": 0, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 1, "total_error": 0}}, "12": {"suite_name": "tutorial/test_secondfixtures.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test1_order_history_title", "rerun": "0"}, "1": {"status": "PASS", "message": "", "test_name": "test2_change_password_title", "rerun": "0"}}, "status": {"total_pass": 2, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "13": {"suite_name": "tutorial/test_skip.py", "tests": {"0": {"status": "SKIP", "message": "('C:\\\\Users\\\\AgnesYeboaaAnsah\\\\Documents\\\\Python_Selenium_Automation\\\\First_Automation_Tutorial\\\\tutorial\\\\test_skip.py', 10, 'Skipped')\n", "test_name": "test_skip1", "rerun": "0"}, "1": {"status": "SKIP", "message": "('C:\\\\Users\\\\AgnesYeboaaAnsah\\\\Documents\\\\Python_Selenium_Automation\\\\First_Automation_Tutorial\\\\tutorial\\\\test_skip.py', 15, 'Skipped: this is yet to be deployed to the test environment')\n", "test_name": "test_skip2", "rerun": "0"}, "2": {"status": "SKIP", "message": "('C:\\\\Users\\\\AgnesYeboaaAnsah\\\\Documents\\\\Python_Selenium_Automation\\\\First_Automation_Tutorial\\\\tutorial\\\\test_skip.py', 22, 'Skipped: unconditional skip')\n", "test_name": "test_skip3", "rerun": "0"}, "3": {"status": "PASS", "message": "", "test_name": "test_skip4", "rerun": "0"}}, "status": {"total_pass": 1, "total_skip": 3, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 0, "total_error": 0}}, "14": {"suite_name": "tutorial/test_stoptest.py", "tests": {"0": {"status": "PASS", "message": "", "test_name": "test_square", "rerun": "0"}, "1": {"status": "FAIL", "message": "   assert 40 == (20 / 20)\n", "test_name": "test_divide", "rerun": "0"}, "2": {"status": "FAIL", "message": "   assert 400 == (20 ** 3)\n", "test_name": "test_cube", "rerun": "0"}, "3": {"status": "PASS", "message": "", "test_name": "test_sum", "rerun": "0"}}, "status": {"total_pass": 2, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 2, "total_error": 0}}}}, "date": "March 21, 2025", "start_time": 1742563191.6441975, "total_suite": 15, "status": "FAIL", "status_list": {"pass": "30", "fail": "18", "skip": "3", "error": "0", "xpass": "2", "xfail": "3", "rerun": "0"}, "total_tests": "56"}